src/ui/Templates/Template.ts

```diff
@@ -124,30 +127,34 @@ export class Template implements ITemplateProperties {
-      // Condition (as a function) is eval'ed, first
-      if (this.condition != null && this.condition(object)) {
-        this.logger.trace('Template was loaded because condition was :', this.condition, object);
-        return this.dataToString(object);
-      }
-      // Condition (as a string) is parsed, if available.
-      if (this.conditionToParse != null && TemplateConditionEvaluator.evaluateCondition(this.conditionToParse, object, instantiateOptions.responsiveComponents)) {
-        this.logger.trace('Template was loaded because condition was :', this.conditionToParse, object);
-        return this.dataToString(object);
-      }
-      // fieldsToMatch is yet another fallback that allows to specify if a template should be loaded.
-      if (this.fieldsToMatch != null && TemplateFieldsEvaluator.evaluateFieldsToMatch(this.fieldsToMatch, object)) {
-        this.logger.trace('Template was loaded because condition was :', this.fieldsToMatch, object);
-        return this.dataToString(object);
-      }
-      // If there is no condition at all, this means "true"
-      if (this.condition == null && this.conditionToParse == null && this.fieldsToMatch == null) {
-        this.logger.trace('Template was loaded because there was *NO* condition', this.condition, object);
-        return this.dataToString(object);
+      try {
+        this.logger.trace('Evaluating template ...');
+        // Condition (as a function) is eval'ed, first
+        if (this.condition != null && this.condition(object)) {
+          this.logger.trace('Template was loaded because condition was :', this.condition, object);
+          return this.dataToString(object);
+        }
+        // Condition (as a string) is parsed, if available.
+        if (this.conditionToParse != null && TemplateConditionEvaluator.evaluateCondition(this.conditionToParse, object, instantiateOptions.responsiveComponents)) {
+          this.logger.trace('Template was loaded because condition was :', this.conditionToParse, object);
+          return this.dataToString(object);
+        }
+        // fieldsToMatch is yet another fallback that allows to specify if a template should be loaded.
+        if (this.fieldsToMatch != null && TemplateFieldsEvaluator.evaluateFieldsToMatch(this.fieldsToMatch, object)) {
+          this.logger.trace('Template was loaded because condition was :', this.fieldsToMatch, object);
+          return this.dataToString(object);
+        }
+        // If there is no condition at all, this means "true"
+        if (this.condition == null && this.conditionToParse == null && this.fieldsToMatch == null) {
+          this.logger.trace('Template was loaded because there was *NO* condition', this.condition, object);
+          return this.dataToString(object);
+        }
+      } catch (e) {
+        new Logger(this).error(l('CannotInstanciateTemplate', (this.element.id != '' ? this.element.id : 'ID undefined')), e.message, this.element);
```

Comment:
In order to have a more insightful log message, I think we should have a `protected` method that returns information to identify the given template. This method could be overriden in classes that inherit from `Template`. ie: Underscore templates could output the HTML element that the template is associated with.